### PR TITLE
Create data

### PR DESCRIPTION
**Pull request by LumisD**
Move a logic to PuzzleLogicRepository
Please review

### CODE REVIEW

File: app/src/main/java/com/lumisdinos/gameoffifteen/data/PuzzleLogicRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class PuzzleLogicRepositoryImpl @Inject constructor(
+    private val gameRepository: GameRepository
+) : PuzzleLogicRepository {
+
+    override var gameStateLive = MutableLiveData<GameState>()
```
Comment (alexeyw) [inline app/src/main/java/com/lumisdinos/gameoffifteen/data/PuzzleLogicRepositoryImpl.kt]: MutableLiveData on data layer?

File: app/src/main/java/com/lumisdinos/gameoffifteen/ui/dialog/SimpleAlertDialog.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package com.lumisdinos.gameoffifteen.ui.dialog
+
+import android.app.AlertDialog
+import android.content.Context
+import com.lumisdinos.gameoffifteen.R
+import com.lumisdinos.gameoffifteen.domain.repos.PuzzleLogicRepository.Companion.ACTION_CONGRATULATIONS
+
+fun getAlertDialog(
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/lumisdinos/gameoffifteen/ui/dialog/SimpleAlertDialog.kt]: Why not using normal alert in Fragment? It looks much cleaner
```
context?.let {
            MaterialAlertDialogBuilder(it)
                .setTitle("Title")
                .setMessage("Message")
                .setPositiveButton(getString(R.string.delete)) { _, _ ->
                  positiveClick()
                }
                .setNegativeButton(getString(R.string.cancel), null)
                .show()
        }
```

Comment (vladislav-v-rusakov) [reply app/src/main/java/com/lumisdinos/gameoffifteen/ui/dialog/SimpleAlertDialog.kt]: Or move it to base fragment 
base fragment

Comment (vladislav-v-rusakov) [reply app/src/main/java/com/lumisdinos/gameoffifteen/ui/dialog/SimpleAlertDialog.kt]: it's just suggestions, your solution is not bad

Comment (LumisD) [reply app/src/main/java/com/lumisdinos/gameoffifteen/ui/dialog/SimpleAlertDialog.kt]: I did it on your MaterialAlertDialogBuilder suggestion as in this particular the dialog is simple one

File: app/src/main/java/com/lumisdinos/gameoffifteen/data/PuzzleLogicRepositoryImpl.kt (hunk 1 above)
Comment (LumisD) [reply app/src/main/java/com/lumisdinos/gameoffifteen/data/PuzzleLogicRepositoryImpl.kt]: I replaced LiveData with Flow

**Review by vladislav-v-rusakov — APPROVED** [review]

**Pull request merged**